packages/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts

```diff
@@ -61,30 +62,41 @@ interface SerializedRow {
   mergedCells: { leftIndex: number; rightIndex: number }[];
 }
 
-export const serializeRow = (
+/**
+ * FIXME: This function mutates the colspan info, but colspan info assumes that the columns
+ * passed to it are always consistent. In this case, the exported columns may differ from the
+ * actual rendered columns.
+ * The caller of this function MUST call `resetColSpan()` before and after usage.
+ */
+export const serializeRowUnsafe = (
   id: GridRowId,
   columns: GridStateColDef[],
-  api: GridPrivateApiPremium,
+  apiRef: React.MutableRefObject<GridPrivateApiPremium>,
```

Comment:
Nitpick: if this change was done to use the `gridHasColSpanSelector` selector, `api` could also work like this:
```ts
gridHasColSpanSelector(api.state, api.instanceId)
```

Reply:
It was. I prefer `api` to `apiRef`, but I like consistency more, so I'd leave it as it is now.